pkgs/servers/nosql/janusgraph/unwrapped.nix

```diff
@@ -0,0 +1,93 @@
+{ stdenv
+, lib
+, maven
+, jdk
+, protobuf
+, protoc-gen-grpc-java
+, unzip
+, version
+, src
+, srcIsBinary ? false
+, outputHash ? null
+}:
+
+let
+  pname = "janusgraph-unwrapped";
+  fetchedMavenDeps = stdenv.mkDerivation {
+    pname = "${pname}-maven-deps";
+    inherit version src;
+
+    nativeBuildInputs = [
+      (maven.override { inherit jdk; })
+      jdk
+      protobuf
+      protoc-gen-grpc-java
+    ];
+
+    patchPhase = ''
+      runHook prePatch
+      sed -i 's|<protocArtifact>.*</protocArtifact>|<protocExecutable>${protobuf}/bin/protoc</protocExecutable>|g; /<pluginArtifact>/d' janusgraph-grpc/pom.xml
+      runHook postPatch
+    '';
+
+    buildPhase = ''
+      runHook preBuild
+      mvn package \
+        -Dmaven.repo.local=$out \
+        -Pjanusgraph-release \
+        -Dmaven.test.failure.ignore=true \
+        -Dgpg.skip=true \
+        -Dadditionalparam=-Xdoclint:none \
+        -DadditionalJOption=-Xdoclint:none
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+      find $out -type f \
+        -name \*.lastUpdated -or \
+        -name resolver-status.properties -or \
+        -name _remote.repositories \
+        -delete
+      runHook postInstall
+    '';
+
+    dontFixup = true;
+    dontConfigure = true;
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    inherit outputHash;
+  };
+in
+stdenv.mkDerivation ({
+  inherit pname version src;
+
+  installPhase = ''
+    cp -r $src $out
```

Comment:
Can we use ln or mv to speed things up?

Reply:
As `$src` is read-only, both `mv` and `ln` fail; `mv` for obvious reasons, and `ln` because the `fixupPhase` tries to change permissions on the `$out` files.
Do you think it's a good idea to disable the `fixupPhase`?

Reply:
fetchurl and fetchzip both support pname and version.